### ISSUE TITLE
Marathon-bigip-ctlr:1.1.0 stops working when F5_0_BIND_ADDR is duplicated

### ISSUE THREAD

**Issue by frankmesen** [F5Networks/marathon-bigip-ctlr#241]
#### Description
For bug: If a new application uses a F5_0_BIND_ADDR that is already in use, the Marathon BigIP Controller stops processing requests.
How to reproduce the bug, create an application and assign a value to F5_0_BIND_ADDR that is already in use by another application.
I would expect the Marathon BigIP Controller to skip the new configuration, alert about it and continue processing requests.

#### Mesos Version
Version 1.2.3

#### Marathon Version
Version: 1.4.7

#### Controller Version
Version: 1.1.0

#### BIG-IP Version
12.1.1 HF1

#### Diagnostic Information
<Configuration files, error messages, logs>
2017-12-01 23:55:17,419 f5_cccl.resource.resource: ERROR   : HTTP error(400): CCCL resource(ApiVirtualServer) /space/app1_80.
2017-12-01 23:55:17,420 f5_cccl.service.manager: ERROR   : F5CcclResourceRequestError - 400 Unexpected Error: Bad Request for uri: https://loadbalancer_dns/mgmt/space/ltm/virtual/application/
Text: u'{"code":400,"message":"01070333:3: Virtual Server /space/app1_80 illegally shares destination address, source address, service port, ip-protocol, and vlan with Virtual Server //space/app1_2_80.","errorStack":[],"apiError":3}'
2017-12-01 23:55:17,420 f5_cccl.service.manager: ERROR   : Resource /space/app1_80 update error, requeuing task...
https://support.f5.com/csp/article/K60974137

**Comment by frankmesen** [comment]
Hi, 
I did some testing and changed:
if incomplete:
                        # Some retryable error occurred),
                        # do a reset so that we try again
                        self.retry_backoff(self.reset_from_tasks)
to:
if incomplete:
                        # Some retryable error occurred),
                        # do a reset so that we try again
                        logger.error("Unable to update the configuration, please check if there are any invalid data")
                        # comment out the retry_backoff to not disrupt the any updates
                        #self.retry_backoff(self.reset_from_tasks)
Of the marathon-bigip-ctlr.py, the result of that change seems to be what we need.
Basically, the conflict doesn't cause any real impact since the F5_0_BIND_ADDR doesn't get updated.
I imagine that there is a reason for running retry_backoff but I wonder if it's worth to slow down the component for some configuration issue that hasn't even been applied thanks to the module validation.py if I am not mistaken.
Please let me know what you think about it.

Regards,

**Comment by edarzins** [comment]
Frank,
Your observations suggest that the retry backoff isn't behaving correctly -- the purpose of which is to  keep retrying in case of a config error, but using a progressive backoff so we don't retry too often.

I will reproduce this test case and let you know what I find. Thanks!

**Comment by frankmesen** [comment]
Actually, it is working as expected.
Our issue is that behavior because a bad neighbor can affect the whole cluster.

**Comment by edarzins** [comment]
I was able to reproduce this situation. However, the marathon-bigip-ctlr doesn't actually stop working. The progressive backoff will eventually reach a maximum of 128 seconds (if the error persists), so at that point the controller could take up to 128 seconds to respond to a Marathon event.

The purpose of the retry backoff is to retry errors that might be transient in nature instead of ignoring them.

**Comment by frankmesen** [comment]
Yes, that actually my previous message.
It looks like it's design and I don't disagree with it, it looks a good practice to protect the LB.
We will make changes to cope with this behavior.
Thank you very much for your time and help.
Regards,
Frank
